src/getSourceInfo.js

```diff
@@ -1,21 +1,25 @@
 const queryString = require("query-string");
+const assert = require('assert');
 const helpers = require("./getDatasetHelpers");
 
 /**
  * Prototype implementation.
  */
 const getSourceInfo = async (req, res) => {
   const query = queryString.parse(req.url.split('?')[1]);
-  if (!query.prefix) {
-    throw new Error("No prefix defined");
+  try {
+    assert(query.prefix);
+  } catch {
+    return res.status(400).send("No prefix defined");
   }
+
   const {source} = helpers.splitPrefixIntoParts(query.prefix);
 
   let sourceInfo;
   try {
     sourceInfo = await source.getInfo();
   } catch (err) {
-    return helpers.handleError(res, `No source info available`, err.message);
+    return helpers.handle500Error(res, `No source info available`, err.message);
```

Comment:
If we're tidying up the HTTP status codes, then this should be

```
return helpers.handle500Error(res, `Error processing source info`, err.message);
```

(I initially thought `204` (no content) would be better, but reading the code of the `getInfo()` methods, they all seem to return an object if there's no available source info, rather than throwing an error.)